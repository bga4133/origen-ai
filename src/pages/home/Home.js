import React, { useEffect, useState } from "react";
import { SimulationItem } from "../../components/simulation/SimulationItem";
import { MachinesItem } from "../../components/machines/MachinesItem";
import { Filter } from "../../components/Filter/Filter";
import { FormSimulation } from "../../components/formSimulation/FormSimulation";
import { HeaderApp } from "../../components/headerApp/HeaderApp";
import { Loader } from "../../components/Loader/Loader";

export const Home = () => {
  // states
  const [dataApi, setDataApi] = useState([]);
  const [machinesData, setMachinesData] = useState([]);
  const [filtered, setFiltered] = useState([]);
  const [isLoading, setIsLoading] = useState(false);
  const [activeButton, setActiveButton] = useState(0);
  const getSimulationList = async () => {
    setIsLoading(true);
    try {
      const response = await fetch("http://localhost:3004/simulations");
      const machines = await fetch("http://localhost:3004/machines");
      const data = await response.json();
      const machinesResponse = await machines.json();
      const [machineResult] = machinesResponse;
      setMachinesData(machineResult.results);
      const sortData = data.sort((a, b) => (a.created > b.created ? 1 : -1));
      setDataApi(sortData);
      setFiltered(sortData);
      setIsLoading(false);
    } catch (error) {
      setDataApi(null);
      setIsLoading(false);
      throw new Error("Oppps error");
    }
  };

  useEffect(() => {
    getSimulationList();
  }, []);

  return (
    <div>
      <HeaderApp />
      <Filter
        dataApi={dataApi}
        setFiltered={setFiltered}
        activeButton={activeButton}
        setActiveButton={setActiveButton}
      />
      <div
        className="grid grid-cols-1 gap-3  content-around md:grid-cols-2 lg:grid-cols-3 "
        data-testid="imgId"
      >
        {isLoading && <Loader className="ml-60" />}
        {filtered.map(({ id, label, modified, status, created }) => (
          <SimulationItem
            key={id}
            label={label}
            modified={modified}
            status={status}
            created={created}
          />
        ))}
        {machinesData.map(({ id, info_url, label }) => (
          <MachinesItem key={id} info_url={info_url} label={label} />
        ))}
      </div>
      {dataApi == null && (
        <div className="no__content">
          <h1 className="no__content__title">Oppsss Error</h1>
          <h2>Try Again</h2>
        </div>
      )}
      <FormSimulation setFiltered={setFiltered} />
    </div>
  );
};
