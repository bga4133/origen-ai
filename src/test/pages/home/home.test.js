import { render, screen, waitFor } from "@testing-library/react";
import { Home } from "../../../pages/home/Home";

describe("home page test <Home />", () => {
  it("render the component", async () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
  it("shouldn't show the div with the class container if data is not loaded", async () => {
    const { container } = render(<Home />);
    const containerDiv = container.querySelector("container");
    expect(containerDiv).toBeFalsy();
  });
  it("should show the data after load the page ", async () => {
    const { container } = render(<Home />);
    const containerDiv = container.querySelector("container");
    await waitFor(
      () => {
        expect(screen.getByTestId("imgId")).toBeInTheDocument();
      },
      { timeout: 4000 }
    );
  });
});
