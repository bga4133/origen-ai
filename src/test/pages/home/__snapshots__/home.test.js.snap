// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home page test <Home /> render the component 1`] = `
<div>
  <div>
    <div>
      <header>
        <nav
          class="bg-white border-gray-200 px-4 lg:px-6 py-2.5 dark:bg-gray-800"
        >
          <div
            class="flex flex-wrap justify-between items-center mx-auto max-w-screen-xl"
          >
            <a
              class="flex items-center"
              href="https://flowbite.com"
            >
              <span
                class="self-center text-xl font-semibold whitespace-nowrap dark:text-white"
              >
                Origen Ai
              </span>
            </a>
            <div
              class="flex items-center lg:order-2"
            >
              <button
                aria-controls="mobile-menu-2"
                aria-expanded="false"
                class="inline-flex items-center p-2 ml-1 text-sm text-gray-500 rounded-lg lg:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600"
                data-collapse-toggle="mobile-menu-2"
                type="button"
              >
                <span
                  class="sr-only"
                >
                  Open main menu
                </span>
                <svg
                  class="w-6 h-6"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
                    fill-rule="evenodd"
                  />
                </svg>
                <svg
                  class="hidden w-6 h-6"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
                    fill-rule="evenodd"
                  />
                </svg>
              </button>
            </div>
            <div
              class="hidden justify-between items-center w-full lg:flex lg:w-auto lg:order-1"
              id="mobile-menu-2"
            >
              <ul
                class="flex flex-col mt-4 font-medium lg:flex-row lg:space-x-8 lg:mt-0"
              >
                <li>
                  <p
                    aria-current="page"
                    class="block py-2 pr-4 pl-3 text-white rounded bg-primary-700 lg:bg-transparent lg:text-primary-700 lg:p-0 dark:text-white"
                    rel="noreferrer"
                  >
                    Home
                  </p>
                </li>
              </ul>
            </div>
          </div>
        </nav>
      </header>
    </div>
    <div
      class="mt-5"
    >
      <button
        class="mr-4 ml-5 center inline-flex items-center py-2 px-3 text-sm font-medium text-center text-white bg-blue-700 rounded-lg hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
      >
        All
      </button>
      <button
        class="mr-4  inline-flex items-center py-2 px-3 text-sm font-medium text-center text-white bg-blue-700 rounded-lg hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
      >
        Pending
      </button>
      <button
        class="mr-4 inline-flex items-center py-2 px-3 text-sm font-medium text-center text-white bg-blue-700 rounded-lg hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
      >
        Running
      </button>
      <button
        class="mr-4 inline-flex items-center py-2 px-3 text-sm font-medium text-center text-white bg-blue-700 rounded-lg hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
      >
        Finished
      </button>
    </div>
    <div
      class="grid grid-cols-1 gap-3  content-around md:grid-cols-2 lg:grid-cols-3 "
      data-testid="imgId"
    >
      <div
        class="loader mt-5"
        role="status"
      >
        <svg
          aria-hidden="true"
          class="mr-2 w-8 h-8 text-gray-200 animate-spin dark:text-gray-600 fill-blue-600"
          fill="none"
          viewBox="0 0 100 101"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z"
            fill="currentColor"
          />
          <path
            d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z"
            fill="currentFill"
          />
        </svg>
        <span
          class="sr-only"
        >
          Loading...
        </span>
      </div>
    </div>
    <form
      class="ml-1  w-50 md:ml-6 mt-10 lg:ml-6 mt-10   mt-10 w-full max-w-lg content-center"
    >
      <div
        class="flex flex-wrap -mx-3 mb-6"
      >
        <div
          class="w-full md:w-1/2 px-3 mb-6 md:mb-0"
        >
          <label
            class="block uppercase tracking-wide text-gray-700 text-xs font-bold mb-2"
            for="grid-last-name"
          >
            Label
          </label>
          <input
            class="appearance-none block w-full bg-gray-200 text-gray-700 border border-gray-200 rounded py-3 px-4 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
            id="grid-last-name"
            placeholder="Label"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="w-full md:w-1/2 px-3"
        >
          <label
            class="block uppercase tracking-wide text-gray-700 text-xs font-bold mb-2"
            for="grid-last-name"
          >
            Status
          </label>
          <input
            class="appearance-none block w-full bg-gray-200 text-gray-700 border border-gray-200 rounded py-3 px-4 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
            id="grid-last-name"
            placeholder="status"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="flex flex-wrap -mx-3 mb-6"
      >
        <div
          class="w-full px-3"
        >
          <label
            class="block uppercase tracking-wide text-gray-700 text-xs font-bold mb-2"
            for="grid-password"
          >
            Created
          </label>
          <input
            class="appearance-none block w-full bg-gray-200 text-gray-700 border border-gray-200 rounded py-3 px-4 mb-3 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
            id="grid-password"
            placeholder="created"
            required=""
            type="date"
            value=""
          />
        </div>
      </div>
      <div
        class="flex flex-wrap -mx-3 mb-6"
      >
        <div
          class="w-full px-3"
        >
          <label
            class="block uppercase tracking-wide text-gray-700 text-xs font-bold mb-2"
            for="grid-password"
          >
            Modified
          </label>
          <input
            class="appearance-none block w-full bg-gray-200 text-gray-700 border border-gray-200 rounded py-3 px-4 mb-3 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
            id="grid-password"
            placeholder="Modified"
            required=""
            type="date"
            value=""
          />
          <button
            class="inline-flex items-center py-2 px-3 text-sm font-medium text-center text-white bg-blue-700 rounded-lg hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800"
          >
            Add Simulation
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
